src/routeConfig/index.tsx

```diff
@@ -1,58 +1,49 @@
-import Login from 'components/Login';
-import Permission from 'components/Permission';
 import React from 'react';
-import { Route, RouteProps } from 'react-router-dom';
-import { getUser } from 'utils/auth';
+import Login from 'components/Login';
+import Dashboard from 'components/Dashboard';
+import Admin from 'layout/Admin';
+import Users from 'components/Users';
+import UserForm from 'components/UserForm';
+import Route from './Route';
 import { Role } from 'utils/types';
-import { routes } from './routes';
-interface Props extends RouteProps {
-   roles?: Role[];
-   layout: React.ComponentType<any>;
-}
-
-function RouterConfig({ layout: Layout }: Props) {
-   const currentUser = getUser();
-   if (currentUser && currentUser.email) {
-      return routes.map((route, index) => {
-         const { path, exact, component, roles, withLayout } = route;
 
-         if (roles && roles.includes(currentUser.role)) {
-            if (withLayout) {
-               return (
-                  <Layout>
-                     <Route
-                        key={index}
-                        path={path}
-                        exact={exact}
-                        component={component}
-                     />
-                  </Layout>
-               );
-            }
-            return (
-               <Layout>
-                  <Route
-                     key={index}
-                     path={path}
-                     exact={exact}
-                     component={component}
-                  />
-               </Layout>
-            );
-         }
-         return (
-            <Layout>
-               <Route
-                  key={index}
-                  path={path}
-                  exact={exact}
-                  component={Permission}
-               />
-            </Layout>
-         );
-      });
-   }
-   return <Route key="login" path="/" component={Login} />;
-}
+const routes = [
+   <Route key="login" exact path="/login" component={Login} />,
+   <Route
+      withAuth
+      layout={Admin}
+      roles={[Role.Admin]}
+      path={`/users/create`}
+      component={() => <UserForm isCreate={true} />}
+      exact
+      key="Userform"
+   />,
+   <Route
+      withAuth
+      layout={Admin}
+      roles={[Role.Admin]}
+      path={`/users/update/:id`}
+      component={() => <UserForm isCreate={false} />}
```

Comment:
```suggestion
     component={() => {
         const params = useParams()
         if (params['id'] && Number(params['id'])) {
            const id = Number(params['id'])
            return <UserForm id={id} />;  
         } else {
            return <NotFoundPage />
         }
      }}
```

Nên lấy data param ở đây